rust/fury/src/meta/meta_string.rs

```diff
@@ -0,0 +1,488 @@
+        let total_bits: usize = data.len() * 8;
+        let strip_last_char = (data[0] & 0x80) != 0;
+        let bit_mask: usize = 0b11111;
+        let mut bit_index = 1;
+        while bit_index + 5 <= total_bits && !(strip_last_char && (bit_index + 2 * 5 > total_bits))
+        {
+            let byte_index = bit_index / 8;
+            let intra_byte_index = bit_index % 8;
+            let char_value: usize = if intra_byte_index > 3 {
+                ((data[byte_index] as usize) << 8
+                    | if byte_index + 1 < data.len() {
+                        data.get(byte_index + 1).cloned().unwrap() as usize & 0xFF
+                    } else {
+                        0
+                    })
+                    >> (11 - intra_byte_index)
+                    & bit_mask
+            } else {
+                (data[byte_index] as usize) >> (3 - intra_byte_index) & bit_mask
+            };
+            bit_index += 5;
+            decoded.push(self.decode_lower_special_char(char_value as u8)?);
+        }
+        Ok(decoded)
+    }
+
+    fn decode_lower_upper_digit_special(&self, data: &[u8]) -> Result<String, Error> {
+        let mut decoded = String::new();
+        let num_bits = data.len() * 8;
+        let strip_last_char = (data[0] & 0x80) != 0;
+        let mut bit_index = 1;
+        let bit_mask: usize = 0b111111;
+
+        while bit_index + 6 <= num_bits && !(strip_last_char && (bit_index + 2 * 6 > num_bits)) {
+            let byte_index = bit_index / 8;
+            let intra_byte_index = bit_index % 8;
+            let char_value: usize = if intra_byte_index > 2 {
+                ((data[byte_index] as usize) << 8
+                    | if byte_index + 1 < data.len() {
+                        data.get(byte_index + 1).cloned().unwrap() as usize & 0xFF
+                    } else {
+                        0
+                    })
+                    >> (10 - intra_byte_index)
+                    & bit_mask
+            } else {
+                (data[byte_index] as usize) >> (2 - intra_byte_index) & bit_mask
+            };
+            bit_index += 6;
+            decoded.push(self.decode_lower_upper_digit_special_char(char_value as u8)?);
+        }
+        Ok(decoded)
+    }
+
+    fn decode_lower_special_char(&self, char_value: u8) -> Result<char, Error> {
+        match char_value {
+            0..=25 => Ok((b'a' + char_value) as char),
+            26 => Ok('.'),
+            27 => Ok('_'),
+            29 => Ok('|'),
+            _ => Err(Error::InvalidLowerSpecialValue { value: char_value }),
+        }
+    }
+
+    fn decode_lower_upper_digit_special_char(&self, char_value: u8) -> Result<char, Error> {
+        match char_value {
+            0..=25 => Ok((b'a' + char_value) as char),
+            26..=51 => Ok((b'A' + char_value - 26) as char),
+            52..=61 => Ok((b'0' + char_value - 52) as char),
+            62 => Ok(self.special_char1),
+            63 => Ok(self.special_char2),
```

Comment:
`special_char1 and special_char2 `  are only used for Java. Java requires three specific characters:  `._$`, but only  62 and 63 are available. In Xlang, this logic is unnecessary because you can't determine the meanings of the 62 and 63 when decoding binary.

Reply:
This will be helpful for you.  `https://fury.apache.org/docs/specification/fury_xlang_serialization_spec#encoding-algorithms`

Reply:
My fault. So I will remove these two variables.
However, I have a question. Besides Java, I also referenced the Python PR [#1702](https://github.com/apache/fury/pull/1702), but that python source code still contains `special_char1` and `special_char2`. This is why I kept them in Rust as well. But Python also doesn't support `$` in field name, right?

Reply:
> My fault. So I will remove these two variables. However, I have a question. Besides Java, I also referenced the Python PR [#1702](https://github.com/apache/fury/pull/1702), but that python source code still contains `special_char1` and `special_char2`. This is why I kept them in Rust as well. But Python also doesn't support `$` in field name, right?



> My fault. So I will remove these two variables. However, I have a question. Besides Java, I also referenced the Python PR [#1702](https://github.com/apache/fury/pull/1702), but that python source code still contains `special_char1` and `special_char2`. This is why I kept them in Rust as well. But Python also doesn't support `$` in field name, right?

The  `Encoding Flag` is determined by the function `computeStatistics`.  
The `Encoding Flag` for a  string that containing `$`  would be set to `UTF8` or  `LOWER_SPECIAL`.

I noticed that `special_char1` and `special_char2` were deleted in the commit `https://github.com/apache/fury/pull/1702/commits/266e4e8031bd38b5263b134d5d8a73b78c401a43`